crypto/params.c

```diff
@@ -887,3 +891,20 @@ OSSL_PARAM OSSL_PARAM_construct_end(void)
 
     return end;
 }
+
+int OSSL_PARAM_used(const OSSL_PARAM *p)
+{
+    return (p->flags & OSSL_PARAM_USED) != 0;
+}
+
+void OSSL_PARAM_set_unused(OSSL_PARAM *p)
+{
+    if (p != NULL)
+        p->flags &= ~OSSL_PARAM_USED;
+}
+
+void OSSL_PARAM_set_all_unused(OSSL_PARAM *params)
```

Comment:
I think this should take an "int used" flag also.

Reply:
A set all used defeats the purpose of this.